src/app/advertisement/advertisement.component.ts

```diff
@@ -28,6 +28,8 @@ export class AdvertisementComponent implements OnInit {
           this.getAdvertisements(this.advertisementService.getAllAdvertisements);
         } else {
           this.getAdvertisements(this.searchAdvertisementService.searchAdvertisementByTitle, queryParam.title);
+          this.getAdvertisements(this.searchAdvertisementService.searchAdvertisementByTag, queryParam.tag);
```

Comment:
It will not work as you expect. `this.getAdvertisements` replaces all advertisements will the content it retrieves from the API. Hence, you will see the results of the one which finishes the latest, regardless of the params.
You either have to merge the content, if you support searching by different parameters (which is not the case with the current advanced search), or write your `getAdvertisements` calls as if-else (or switch-case) cases depending on the query params you get.